src/node_snapshotable.cc

```diff
@@ -1,16 +1,181 @@
 
 #include "node_snapshotable.h"
+#include <iostream>
+#include <sstream>
 #include "base_object-inl.h"
 #include "debug_utils-inl.h"
+#include "env-inl.h"
+#include "node_errors.h"
+#include "node_external_reference.h"
 #include "node_file.h"
+#include "node_internals.h"
+#include "node_main_instance.h"
 #include "node_v8.h"
+#include "node_v8_platform-inl.h"
 
 namespace node {
 
+using v8::Context;
+using v8::HandleScope;
+using v8::Isolate;
 using v8::Local;
 using v8::Object;
 using v8::SnapshotCreator;
 using v8::StartupData;
+using v8::TryCatch;
+using v8::Value;
+
+template <typename T>
+void WriteVector(std::ostringstream* ss, const T* vec, size_t size) {
+  for (size_t i = 0; i < size; i++) {
+    *ss << std::to_string(vec[i]) << (i == size - 1 ? '\n' : ',');
+  }
+}
```

Comment:
I don't don't get why we're not using a reference here — is it just stylistic, or is there a performance reason, or what?

Reply:
I think when we need a mutable parameter we usually just go with pointers instead of references.

Reply:
Actually it's in the style guide https://github.com/nodejs/node/blob/master/doc/guides/cpp-style-guide.md#avoid-non-const-references

Reply:
Ah, that's very clear. Thank you!